### DISCUSSION TITLE
Essential boundary condition u.n = 0 on disc.

### DISCUSSION THREAD

**Discussion by ethrelfall** [firedrakeproject/firedrake#2552] [Firedrake support]
This should be a simple one - I've got a primal formulation of the vector Laplacian for u on (say) a disc and I want to be able to strongly enforce u.n=0 on the boundary.

I think I need to add a DirichletBC, but what is the interface for doing this, please? 

If I do 

bc = DirichletBC(V, as_vector([0.0, 0.0]), {domain IDs})  ## V is VectorFunctionSpace(mesh, "CG", 1) and mesh is 2D

I get both components of u zero on the boundary, whereas I only want to fix the radial component.

Aside - I think V.sub(0) gets me the Cartesian x-component of V - right?  

Thanks!

**Comment by ksagiyam** [comment]
Yes, `V.sub(0)` is for the Cartesian `x` component.

I'm afraid we currently cannot apply `u.n = g` condition strongly if `n` does not align with the coordinate axis. There was a branch which made this possible, but that branch is not kept up-to-date.

**Reply by ethrelfall** [reply]
Thanks for making this clear.

**Comment by danshapero** [comment]
I know you wanted a more direct approach, but I've had good luck with Nitsche's method -- I wrote about it [here](https://shapero.xyz/posts/nitsches-method-stokes/).

**Reply by ethrelfall** [reply]
A turnkey answer would have been nice but that page looks very good, plus I'm also interested in fluid mechanics implemented in Firedrake, so I'll be sure to have a look - thanks!

**Reply by danshapero** [reply]
Yea it's not ideal but it might get the job done. In case you don't like that solution, I think using H(div)-conforming elements like Raviart-Thomas or Brezzi-Douglas-Marini can also accomplish the same thing. I haven't tried it but @colinjcotter has recommended it to me before so you could ask him.

**Reply by colinjcotter** [reply]
Yes, just set a BC with zero on an Hdiv space just results in setting the normal component.
